juicy-highlight.html

```diff
@@ -1,294 +1,298 @@
-            var maskSvg = calcMask(availableWidth, availableHeight, maskElement.focused, maskElement.strokeOffset);
-            var rectSvg = "";
-            var rect = focusRect.firstChild;
-            var hidden = maskElement.currentState != "shown";
-
-            if (maskElement.focused.length) {
-                for (var i = 0, ilen = maskElement.focused.length; i < ilen; i++) {
-                    rectSvg += calcFocusRect(maskElement.focused[i], maskElement.strokeOffset);
+                if (hidden || maskElement.currentRectSvg != rectSvg) {
+                    rect.setAttribute("d", rectSvg);
+                    maskElement.currentRectSvg = rectSvg;
                 }
-            }
-            else {
-                rectSvg = "M 0 0";
-            }
 
-            if (overlay && (hidden || maskElement.currentMaskSvg != maskSvg)) {
-                var mask = svg.firstChild;
+                var w = availableWidth + "px";
+                var h = availableHeight + "px";
+                var els = [svg, focusRect];
 
-                mask.setAttribute("d", maskSvg);
-                maskElement.currentMaskSvg = maskSvg;
-            }
+                for (var i = 0; i < els.length; i++) {
+                    var el = els[i];
 
-            if (hidden || maskElement.currentRectSvg != rectSvg) {
-                rect.setAttribute("d", rectSvg);
-                maskElement.currentRectSvg = rectSvg;
+                    if (el.style.width != w) {
+                        el.style.width = w;
+                    }
+
+                    if (el.style.height != h) {
+                        el.style.height = h;
+                    }
+                }
             }
+        class JuicyHighlightElement extends HTMLElement {            
+            constructor(self) {
+                self = super(self);
+                self.overlay = false;
+                self.fill = false;
+                self.strokeWidth = 1;
+                self.strokeOffset = 1;
+                self.strokeColor = "#000000";
+                self.focused = null;
+                self.contents = [];
+                self.currentState = null;
+                self.currentMaskSvg = null;
+                self.currentRectSvg = null;
+
+                var attributes = ["overlay", "strokewidth", "strokeoffset", "strokecolor", "fill"];
+
+                for (var i = 0; i < attributes.length; i++) {
+                    if (self.hasAttribute(attributes[i])) {
+                        self.attributeChangedCallback(attributes[i], null, self.attributes[attributes[i]].value);
+                    }
+                }
+            };
+            
+            static get observedAttributes() {
+                return ["overlay", "strokewidth", "strokeoffset", "strokecolor", "fill"]; 
+            }
+            
+            attributeChangedCallback(attrName, oldValue, newValue) {
+                switch (attrName.toLowerCase()) {
```

Comment:
It can be just `switch (attrName) {`.

I cannot find in the spec how is the attribute name normalized in HTML. None of the online examples normalize it inside the callback, so I assume it is the same case as you provided in `observedAttributes`.

As least in `wc-lite.js`  CEv1 polyfill, `attributeChangedCallback` gets exactly the same case that was registered in `observedAttributes`.